### PR TITLE
Update documentation, changelog, release readme

### PR DESCRIPTION
**Pull request by leochr**
**What this PR does / why we need it?**:

- Update documentation, changelog
- Add replicas and expose to default sample in CSV
- Delete old crds, roles and role bindings
- Add initial readme for 0.8.0

**Does this PR introduce a user-facing change?**

- [x] User guide
- [x] `CHANGELOG.md`

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: deploy/releases/0.8.0/readme.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+== Supported Environments
+
+The Runtime Component Operator is tested and built on the following CPU architectures:
+
+* amd64
+
+The operator will run on any cluster that has Kubernetes v1.19 or higher.
+
+== Installation
```
Comment (idlewis) [inline deploy/releases/0.8.0/readme.adoc:15]: Should this section explicitly state 'Installation on OCP' or something similar?

File: doc/user-guide.adoc [hunk 2]
```diff
@@ -78,7 +78,7 @@ Each `RuntimeComponent` CR must at least specify the `applicationImage` paramete
 | `service.nodePort` | Node proxies this port into your service. Please note once this port is set to a non-zero value it cannot be reset to zero.
 | `service.annotations` | Annotations to be added to the service.
 | `service.certificate` | A YAML object that represents a link:++https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1alpha2.CertificateSpec++[Certificate].
-| `service.certificateSecretRef` | A name of a secret that already contains TLS key, certificate and CA to be mounted in the pod.
+| `service.certificateSecretRef` | A name of a secret that already contains TLS key, certificate and CA to be mounted in the pod. The following keys are valid in the secret: `ca.crt`, `tls.crt`, and `tls.key`.
```
Comment (idlewis) [inline doc/user-guide.adoc:81]: Should the struct comments in runtimecomponent_types.go be updated to match this description?

File: doc/user-guide.adoc [hunk 3]
```diff
@@ -120,7 +120,7 @@ Each `RuntimeComponent` CR must at least specify the `applicationImage` paramete
 | `route.termination`   | TLS termination policy. Can be one of `edge`, `reencrypt` and `passthrough`.
 | `route.insecureEdgeTerminationPolicy`   | HTTP traffic policy with TLS enabled. Can be one of `Allow`, `Redirect` and `None`.
 | `route.certificate`  | A YAML object that represents a link:++https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1alpha2.CertificateSpec++[Certificate].
-| `route.certificateSecretRef` | A name of a secret that already contains TLS key, certificate and CA to be used in the route. Also can contain destination CA certificate.
+| `route.certificateSecretRef` | A name of a secret that already contains TLS key, certificate and CA to be used in the route. It can also contain destination CA certificate. The following keys are valid in the secret: `ca.crt`, `destCA.crt`, `tls.crt`, and `tls.key`.
```
Comment (idlewis) [inline doc/user-guide.adoc:123]: Same thing here as for the service.certificateSecretRef

**Comment by leochr** [comment]
@idlewis I'll address the comments via a separate PR. Merging this PR as there were some inquiries about install steps.

**Pull request merged**